From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pierre=20Sandstro=CC=88m?= <pierre.sandstrom@dstny.com>
Date: Thu, 15 Dec 2022 17:24:13 +0100
Subject: Adding external codecs to codec factory.

This patch adds our external codecs (AMR-WB/NB and G.729) to the WebRTC codec factory used by Chromium.

diff --git a/third_party/blink/renderer/platform/peerconnection/audio_codec_factory.cc b/third_party/blink/renderer/platform/peerconnection/audio_codec_factory.cc
index 0165a9bd17c9166389c9f0492cff01b488153ac5..1eefc8d13d2c6ca774db006cdf60302299170606 100644
--- a/third_party/blink/renderer/platform/peerconnection/audio_codec_factory.cc
+++ b/third_party/blink/renderer/platform/peerconnection/audio_codec_factory.cc
@@ -22,6 +22,13 @@
 #include "third_party/webrtc/api/audio_codecs/opus/audio_encoder_multi_channel_opus.h"
 #include "third_party/webrtc/api/audio_codecs/opus/audio_encoder_opus.h"

+#include "third_party/webrtc/mediaengine/codecs/g729/audio_decoder_g729.h"
+#include "third_party/webrtc/mediaengine/codecs/g729/audio_encoder_g729.h"
+#include "third_party/webrtc/mediaengine/codecs/amr/audio_decoder_amr.h"
+#include "third_party/webrtc/mediaengine/codecs/amr/audio_encoder_amr.h"
+#include "third_party/webrtc/mediaengine/codecs/amr/audio_decoder_amrwb.h"
+#include "third_party/webrtc/mediaengine/codecs/amr/audio_encoder_amrwb.h"
+
 namespace blink {

 namespace {
@@ -74,7 +81,10 @@ rtc::scoped_refptr<webrtc::AudioEncoderFactory>
 CreateWebrtcAudioEncoderFactory() {
   return webrtc::CreateAudioEncoderFactory<
       webrtc::AudioEncoderOpus, webrtc::AudioEncoderIsac,
+      mediaengine::AudioEncoderAmrWb,
+      mediaengine::AudioEncoderAmr,
       webrtc::AudioEncoderG722, webrtc::AudioEncoderG711,
+      mediaengine::AudioEncoderG729,
       NotAdvertisedEncoder<webrtc::AudioEncoderL16>,
       NotAdvertisedEncoder<webrtc::AudioEncoderMultiChannelOpus>>();
 }
@@ -83,7 +93,10 @@ rtc::scoped_refptr<webrtc::AudioDecoderFactory>
 CreateWebrtcAudioDecoderFactory() {
   return webrtc::CreateAudioDecoderFactory<
       webrtc::AudioDecoderOpus, webrtc::AudioDecoderIsac,
+      mediaengine::AudioDecoderAmrWb,
+      mediaengine::AudioDecoderAmr,
       webrtc::AudioDecoderG722, webrtc::AudioDecoderG711,
+      mediaengine::AudioDecoderG729,
       NotAdvertisedDecoder<webrtc::AudioDecoderL16>,
       NotAdvertisedDecoder<webrtc::AudioDecoderMultiChannelOpus>>();
 }
